lib/subprocess.js

```diff
@@ -52,11 +96,19 @@ class SubProcess extends EventEmitter {
 
   // spawn the subprocess and return control whenever we deem that it has fully
   // "started"
+  /**
+   *
+   * @param {StartDetector|number?} startDetector
+   * @param {number?} timeoutMs
+   * @param {boolean} detach
+   * @returns {Promise<void>}
+   */
   async start (startDetector = null, timeoutMs = null, detach = false) {
     let startDelay = 10;
 
+    /** @type {StartDetector} */
     const genericStartDetector = function genericStartDetector (stdout, stderr) {
-      return stdout || stderr;
+      return stdout ?? stderr;
```

Comment:
i think in this case we might still want to start if `stdout` is `''` but `stderr` is truthy

Reply:
i.e., if `stdout` is `''` but `stderr` is `process started!`, this function will now return `''`, which will evaluate to false later on, when with the current logic, the function would return `process started!`.

i.e. x 2, i think this might be an unintentional breaking change.

Reply:
k